src/styles/calendar.scss

```
@@ -0,0 +1,50 @@
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  width: $width-cell * 7 + $padding-cal * 2 + $gap * 6;
+  margin: 20px auto 0;
+  gap: $gap;
+  padding: $padding-cal ;
+  box-sizing: border-box;
+
+  @each $day , $num in $days {
+    &--start-#{$day} :first-child {
+      margin-left: ($width-cell + $gap) * $num;
+    }
+  }
+
+  @for $day from 28 to 31 {
+    &--month-#{$day} &__day:nth-child(n + #{$day + 1}) {
+      display: none;
+    }
+  }
+
+  &__day {
+    display: flex;
+    background-color: #eee;
+    width: $width-cell ;
+    height: $height-cell;
+    border: 1px solid grey;
+    box-sizing: border-box;
+    position: relative;
+    transition: transform 0.5s;
+    justify-content: center;
+    align-items: center;
+
+    @for $i from 1 through 31 {
+      &:nth-child(#{$i}):before {
+        color: #000;
+        font-size: 30px;
+        font-family: Arial, Helvetica, sans-serif;
+        content: "#{$i}";
+      }
+    }
```

```suggestion
    &:before {
      color: #000;
      font-size: 30px;
      font-family: Arial, Helvetica, sans-serif;
    }

    @for $i from 1 through 31 {
      &:nth-child(#{$i}):before {
        content: "#{$i}";
      }
    }
```
You don't need to specify these properties to each `:nth-child(#{$i})`. Just once to `:before` selector.
Something like this